File: services/assesment-service/src/main/java/sgptt/assessvc/assesmentservice/entity/Sinodal.java
package sgptt.assessvc.assesmentservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Sinodal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "person_id")
    private Professor professor;

    @Column(name = "is_sinodal", nullable = false, columnDefinition = "boolean default true")
    private Boolean isSinodal;

    @ManyToOne
    @JoinColumn(name = "protocol_id")
    private Protocol protocol;

}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/entity/Sinodal.kt
package com.sgptt.protocolsservice.repository.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
data class Sinodal(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	val id: Long,
	
	@ManyToOne
	@JoinColumn(name = "person_id")
	val professor: Professor,
	
	@Column(name = "is_sinodal", nullable = false, columnDefinition = "boolean default true")
	val isSinodal: Boolean,
	
	@ManyToOne
	@JoinColumn(name = "protocol_id")
	val protocol: Protocol
)
